components/eam/src/physics/crm/sam/crm_variance_transport.F90

```diff
@@ -0,0 +1,368 @@
+
+            ! transform back
+            call rfft1b( nx, 1, fft_out(:), nx, wsave, lensav, work(:), nx, ier )
+            if(ier /= 0) write(0,*) 'ERROR: rfftmb(): VT_filter - backward FFT error ',ier
+
+            ! copy to output
+            do i = 1,nx
+               f_out(icrm,i,j,k) = fft_out(i)
+            end do
+
+         end do
+      end do
+   end do
+
+end subroutine VT_filter
+
+!===============================================================================
+!===============================================================================
+subroutine VT_diagnose(ncrms)
+   !----------------------------------------------------------------------------
+   ! Purpose: Diagnose amplitude for each wavenumber for variance transport
+   !----------------------------------------------------------------------------
+   use crmdims,   only: crm_dx
+   use vars,      only: t,qv,qcl,qci
+   implicit none
+
+   ! interface arguments
+   integer, intent(in) :: ncrms
+
+   ! local variables
+   real(crm_rknd), allocatable :: t_mean(:,:)
+   real(crm_rknd), allocatable :: q_mean(:,:)
+   real(crm_rknd), allocatable :: tmp_t(:,:,:,:)
+   real(crm_rknd), allocatable :: tmp_q(:,:,:,:)
+   real(crm_rknd) :: tmp
+   real(crm_rknd) :: factor_xy
+   integer :: i, j, k, icrm   ! loop iterators
+   !----------------------------------------------------------------------------
+
+   allocate( t_mean( ncrms, nzm ) )
+   allocate( q_mean( ncrms, nzm ) )
+
+   allocate( tmp_t( ncrms, nx, ny, nzm ) )
+   allocate( tmp_q( ncrms, nx, ny, nzm ) )
+
+   factor_xy = 1._crm_rknd/dble(nx*ny)
+
+   !----------------------------------------------------------------------------
+   ! calculate horizontal mean
+   !----------------------------------------------------------------------------
+   !$acc parallel loop collapse(2) async(asyncid)
+   do k = 1,nzm
+      do icrm = 1,ncrms
+         t_mean(icrm,k) = 0.
+         q_mean(icrm,k) = 0.
+         t_vt(icrm,k) = 0.
+         q_vt(icrm,k) = 0.
+      end do
+   end do
+
+   !$acc parallel loop collapse(4) async(asyncid)
+   do k = 1,nzm
+      do j = 1,ny
+         do i = 1,nx
+            do icrm = 1,ncrms
+               !$acc atomic update
+               t_mean(icrm,k) = t_mean(icrm,k) + t(icrm,i,j,k)
+               tmp = qv(icrm,i,j,k)+qcl(icrm,i,j,k)+qci(icrm,i,j,k)
+               !$acc atomic update
+               q_mean(icrm,k) = q_mean(icrm,k) + tmp
+            end do
+         end do
+      end do
+   end do
+
+   !$acc parallel loop collapse(2) async(asyncid)
+   do k = 1,nzm
+      do icrm = 1,ncrms
+         t_mean(icrm,k) = t_mean(icrm,k) * factor_xy
+         q_mean(icrm,k) = q_mean(icrm,k) * factor_xy
+      end do
+   end do
+
+   !----------------------------------------------------------------------------
+   ! calculate anomalies
+   !----------------------------------------------------------------------------
+   if (filter_wn_max>0) then ! use filtered state for anomalies
```

Comment:
This is probably nit-picky, but I think it is better to call these fluctuations rather than anomalies.

Reply:
I don't care either way, so I'll use "fluctuations".